### ISSUE TITLE
sqlfluff: terminated by signal 9 while running testing in CI pipeline

### ISSUE THREAD

**Issue by rp926463-arch** [sqlfluff/sqlfluff#6184]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Trying to use sqlfluff as lint job in CI but raising below error

**Error:**
`xargs: sqlfluff: terminated by signal 9
Cleaning up project directory and file based variables
00:01
/usr/lib64/python3.12/multiprocessing/resource_tracker.py:254: UserWarning: resource_tracker: There appear to be 2 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '
ERROR: Job failed: command terminated with exit code 1`

**config : .sqlfluff_dev**
Using starter config from https://docs.sqlfluff.com/en/2.3.2/configuration.html#starter-config with
`templater = dbt`
everything else is same.

_Below job is trying to get sql file changes in current MR, Number of files i'm trying to lint here are 24 with are having quite simple sql with almost all filesizes < 20000 bytes_

in local machine it running with below benchmarks
<img width="566" alt="image" src="https://github.com/user-attachments/assets/45b93acf-518a-4343-a9aa-0fd827cb2831">

**CI job**
lint job is as below
```shell
lint-test:
  stage: lint-test
  #allow_failure: true
  <<: *pre
  environment:
    name: dev
  tags:
    - warehouse
  rules:
    - if:
        $CI_PIPELINE_SOURCE == "merge_request_event" && $CI_MERGE_REQUEST_TARGET_BRANCH_NAME == 'ci-test'
      changes:
        - models/**/*.sql
  script:
    - echo $CI_DEFAULT_BRANCH
    - git config --global --add safe.directory $CI_PROJECT_DIR
    - git fetch --depth 1 origin $CI_MERGE_REQUEST_DIFF_BASE_SHA
    - CHANGED_SQL_FILES=$(git diff --name-only --diff-filter=d $CI_MERGE_REQUEST_DIFF_BASE_SHA $CI_COMMIT_SHA | grep -E '\.sql$' || true)
    - echo "Changed SQL files $CHANGED_SQL_FILES"
    - |
      if [ -n "$CHANGED_SQL_FILES" ]; then
        echo "SQL files have changed. Running sqlfluff..."
        echo "Linting code..."
        echo $CHANGED_SQL_FILES | xargs sqlfluff lint --config .sqlfluff_dev --bench
      else
        echo "No SQL files have changed. Skipping sqlfluff..."
      fi
``` 

### Expected Behaviour

Lint should complete clean with All finished or with issues

### Observed Behaviour

**Error:**
`xargs: sqlfluff: terminated by signal 9
Cleaning up project directory and file based variables
00:01
/usr/lib64/python3.12/multiprocessing/resource_tracker.py:254: UserWarning: resource_tracker: There appear to be 2 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '
ERROR: Job failed: command terminated with exit code 1`

### How to reproduce

Try to lint on multiple .sql files (preferably > 25)
with above config & in gitlab ci if possible

### Dialect

snowflake

### Version

3.1.0

### Configuration

[sqlfluff]
dialect = snowflake

templater = dbt
exclude_rules = ambiguous.column_count, structure.column_order

max_line_length = 120

processes = 1

[sqlfluff:templater:dbt]
project_dir = ./
profiles_dir = ./
profile = local
target = dev
execute = False

[sqlfluff:indentation]
allow_implicit_indents = True

[sqlfluff:rules:aliasing.table]
aliasing = explicit
[sqlfluff:rules:aliasing.column]
aliasing = explicit

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = lower
[sqlfluff:rules:capitalisation.identifiers]
capitalisation_policy = lower
[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = lower
[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = lower
[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = lower

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
Fixed in #7557
